### PR TITLE
Adding custom error class

### LINKED ISSUE #79: Using a custom error class
This is useful when used together with [pytest-rerunfailures](https://github.com/pytest-dev/pytest-rerunfailures), and it can help to also with `try` context managers.

### PR DESCRIPTION
**Pull request by germa89**
As the title.

Close #79

### CODE REVIEW

**Comment by germa89** [comment]
pre-commit.ci autofix

File: .github/workflows/ci_cd.yml [hunk 1]
```diff
@@ -102,7 +102,7 @@ jobs:
 
       - name: Unit Testing
         run: xvfb-run coverage run -m --source=pytest_pyvista --module pytest --verbose .
-      - uses: codecov/codecov-action@v4
+      - uses: codecov/codecov-action@v3
```
Comment (MatthewFlamm) [inline .github/workflows/ci_cd.yml:105]:
```suggestion
      - uses: codecov/codecov-action@v4
```

File: pytest_pyvista/pytest_pyvista.py [hunk 2]
```diff
@@ -8,6 +8,20 @@
 import pyvista
 
 
+class RegressionError(RuntimeError):
+    """Error when regression does not meet the criteria"""
+
+    def __init__(self, msg):
```
Comment (MatthewFlamm) [inline pytest_pyvista/pytest_pyvista.py:14]: Do we need these __init__ methods?  Can't we instead inherit the parent class method?

**Review by MatthewFlamm — COMMENTED** [review]
Looks good to me. Please use in the tests here too, I think there is only one spot currently.  Also, can we add tests that these get raised?

File: .github/workflows/ci_cd.yml (hunk 1 above)
Comment (germa89) [reply .github/workflows/ci_cd.yml:105]: I think it was deprecated.

File: pytest_pyvista/pytest_pyvista.py (hunk 2 above)
Comment (germa89) [reply pytest_pyvista/pytest_pyvista.py:14]: Maybe you don't needed. I guess it is just an habit.

File: .github/workflows/ci_cd.yml (hunk 1 above)
Comment (germa89) [reply .github/workflows/ci_cd.yml:105]: See https://github.com/pyvista/pytest-pyvista/actions/runs/6254763049/job/16982901626

**Comment by germa89** [comment]
I think I'm done with this!

**Review by MatthewFlamm — APPROVED** [review]
LGTM

**Pull request merged**